src/main/webapp/app/overview/course-exams/course-exams.component.ts

```diff
@@ -1,8 +1,53 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { CourseScoreCalculationService } from 'app/overview/course-score-calculation.service';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs/Subscription';
+import { Exam } from 'app/entities/exam.model';
+import * as moment from 'moment';
 
 @Component({
     selector: 'jhi-course-exams',
     templateUrl: './course-exams.component.html',
     styleUrls: ['./course-exams.scss'],
 })
-export class CourseExamsComponent {}
+export class CourseExamsComponent implements OnInit, OnDestroy {
+    courseId: number;
+    public course: Course | null;
+    private paramSubscription: Subscription;
+    private courseUpdatesSubscription: Subscription;
+
+    constructor(private route: ActivatedRoute, private courseManagementService: CourseManagementService, private courseCalculationService: CourseScoreCalculationService) {}
+
+    /**
+     * subscribe to changes in the course and fetch course by the path parameter
+     */
+    ngOnInit(): void {
+        this.paramSubscription = this.route.parent!.params.subscribe((params) => {
+            this.courseId = parseInt(params['courseId'], 10);
+        });
+
+        this.course = this.courseCalculationService.getCourse(this.courseId);
```

Comment:
Why is this necessary? It gets overridden a few lines later anyways or not?

Reply:
Good catch, I accidentally left it in, I'll remove it with the next commit